t/157-disable-init-by-lua.t

```diff
@@ -0,0 +1,132 @@
+                        -- will be a string contains the 'error'
+                        ngx.log(ngx.ERR, "run init_by_lua")
+                    }
+                }
+            ]]
+
+            assert(os.execute("mkdir -p $html_dir/logs"))
+            local conf_file = "$html_dir/nginx.conf"
+            local f, err = io.open(conf_file, 'w')
+            if not f then
+                ngx.log(ngx.ERR, err)
+                return
+            end
+
+            assert(f:write(conf))
+
+            return conf_file
+        end
+
+        function get_ngx_bin_path()
+            local ffi = require "ffi"
+            ffi.cdef[[char **ngx_argv;]]
+            return ffi.string(ffi.C.ngx_argv[0])
+        end
+    }
+_EOC_
+
+add_block_preprocessor(sub {
+    my $block = shift;
+
+    if (!defined $block->http_config) {
+        $block->set_value("http_config", $http_config);
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+});
+
+log_level("debug");
+no_long_string();
+run_tests();
+
+__DATA__
+
+=== TEST 1: ensure init_by_lua* is not run in signaller process
+--- config
+    location = /t {
+        content_by_lua_block {
+            local conf_file = set_up_ngx_tmp_conf()
+            local nginx = get_ngx_bin_path()
+
+            local cmd = nginx .. " -p $TEST_NGINX_HTML_DIR -c " .. conf_file .. " -s reopen"
+            local p, err = io.popen(cmd)
+            if not p then
+                ngx.log(ngx.ERR, err)
+                return
+            end
+
+            local out, err = p:read('*a')
+            if not out then
+                ngx.log(ngx.ERR, err)
+
+            else
+                ngx.log(ngx.WARN, out)
+            end
+        }
+    }
+--- no_error_log eval
+qr/\[error\] .+ (?!:nginx.pid" failed \(2: No such file or directory\))$/
+
+
+
+=== TEST 2: ensure init_by_lua* is not run when testing Nginx configuration
+--- config
+    location = /t {
+        content_by_lua_block {
+            local conf_file = set_up_ngx_tmp_conf()
+            local nginx = get_ngx_bin_path()
+
+            local cmd = nginx .. " -p $TEST_NGINX_HTML_DIR -c " .. conf_file .. " -t"
+            local p, err = io.popen(cmd)
+            if not p then
+                ngx.log(ngx.ERR, err)
+                return
+            end
+
+            local out, err = p:read('*a')
+            if not out then
+                ngx.log(ngx.ERR, err)
+
+            else
+                ngx.log(ngx.WARN, out)
+            end
+
+            local cmd = nginx .. " -p $TEST_NGINX_HTML_DIR -c " .. conf_file .. " -T"
+            local p, err = io.popen(cmd)
+            if not p then
+                ngx.log(ngx.ERR, err)
+                return
+            end
+
+            local out, err = p:read('*a')
+            if not out then
+                ngx.log(ngx.ERR, err)
+
+            else
+                ngx.log(ngx.WARN, out)
+            end
+        }
+    }
+--- no_error_log
+[error]
```

Comment:
@spacewander Mind renaming this test suite to `159-disable-init-by-lua.t`? See #1394 which already deals with 2 test suites named `157-*.t`.